### PR TITLE
feat(RPS-1004): handle https, add some convenience methods

### PR DESCRIPTION
**Pull request by biering**
- handle https and improve gateway creation
- conform env variables from ADX
- added `tcps()` and `tcp_names()` method
- added `controllers()` method
- make tcp in `get_state()` optional -> returns flange pose
- updated examples accordingly

### CODE REVIEW

**Review by mahsumdemirwb — APPROVED** [review]

**Review by langenhagen — APPROVED** [review]
lgtm

File: examples/03_move_and_set_ios.py [hunk 1]
```diff
@@ -10,17 +10,18 @@
...
+    controllers = await cell.controllers()
 
     # Define a home position
     home_joints = (0, -pi / 4, -pi / 4, -pi / 4, pi / 4, 0)
 
     # Connect to the controller and activate motion groups
-    async with controller:
-        motion_group = controller.motion_group()
+    async with controllers[0] as motion_group:
+        tcp_names = await motion_group.tcp_names()
```
Comment (bompo) [inline examples/03_move_and_set_ios.py:20]: I guess TCP is name only. But would also be nice if its straight an object with the pose offset

File: examples/02_plan_and_execute.py [hunk 2]
```diff
@@ -10,15 +10,19 @@
...
-    controller = await cell.controller("ur")
+    controllers = await cell.controllers()
+    controller = controllers[0]
 
     # Define a home position
     home_joints = (0, -pi / 4, -pi / 4, -pi / 4, pi / 4, 0)
 
     # Connect to the controller and activate motion groups
-    async with controller[0] as mg:
+    async with controller[0] as motion_group:
```
Comment (bompo) [inline examples/02_plan_and_execute.py:20]: `with controller as motion_group`
Thinking about it, it's a bit strange that contoller is an array itself. Maybe a controller.defaultMotionGroup is more explicit

File: nova/core/motion_group.py [hunk 3]
```diff
@@ -44,6 +44,32 @@ def current_motion(self) -> str:
         #    raise ValueError("No MotionId attached. There is no planned motion available.")
         return self._current_motion
 
+    async def plan(self, actions: list[Action], tcp: str) -> wb.models.JointTrajectory:
+        current_joints = await self.joints(tcp=tcp)
+        robot_setup = await self._get_optimizer_setup(tcp=tcp)
```
Comment (bompo) [inline nova/core/motion_group.py:49]: could be cached?

File: nova/core/motion_group.py [hunk 4]
```diff
@@ -176,3 +162,36 @@ async def stop(self):
...
+        Args:
+            tcp (str): The identifier of the tool center point (TCP) to be used for tcp_pose in response. If not set,
+                the flange pose is returned as tcp_pose.
+        """
+        response = await self._api_gateway.motion_group_infos_api.get_current_motion_group_state(
+            cell=self._cell, motion_group=self.motion_group_id, tcp=tcp
+        )
+        return response
+
+    async def joints(self, tcp: str | None = None) -> wb.models.Joints:
```
Comment (bompo) [inline nova/core/motion_group.py:178]: IDK why TCP is offered as a parameter at all? It doesn't make a difference for the joint positions? I would remove it.

File: nova/core/motion_group.py [hunk 5]
```diff
@@ -176,3 +162,36 @@ async def stop(self):
             logger.debug(f"Motion {self.current_motion} stopped.")
         except ValueError as e:
             logger.debug(f"No motion to stop for {self}: {e}")
+
+    async def get_state(self, tcp: str | None) -> wb.models.MotionGroupStateResponse:
```
Comment (bompo) [inline nova/core/motion_group.py:166]: Is there also the option for a state fetching via websocket?

File: nova/gateway.py [hunk 6]
```diff
@@ -76,15 +78,20 @@ def __init__(
         if access_token is None:
             access_token = config("NOVA_ACCESS_TOKEN", default=None)
 
+        if (username is None or password is None) and access_token is None:
```
Comment (bompo) [inline nova/gateway.py:81]: What happens if all three are set? Access_token should have prio.

**Review by bompo — APPROVED** [review]

File: examples/02_plan_and_execute.py (hunk 2 above)
Comment (biering) [reply examples/02_plan_and_execute.py:20]: The index access is just convenience. You can also do:
```python
with controller.motion_group(0) as motion_group:
    ...
```

File: nova/core/motion_group.py (hunk 4 above)
Comment (biering) [reply nova/core/motion_group.py:178]: Yeah true, for joints it should not make a difference.

File: examples/03_move_and_set_ios.py (hunk 1 above)
Comment (biering) [reply examples/03_move_and_set_ios.py:20]: yes, but for that I've added the `.tcps()` method which returns a list of tcp objects

File: nova/gateway.py (hunk 6 above)
Comment (biering) [reply nova/gateway.py:81]: fixed

File: nova/core/motion_group.py (hunk 3 above)
Comment (biering) [reply nova/core/motion_group.py:50]: done

File: nova/core/motion_group.py (hunk 5 above)
Comment (biering) [reply nova/core/motion_group.py]: not yet (besides the auto-generated API), but we should tackle that in a new MR.

**Pull request merged**